router/router.go

```diff
@@ -1819,6 +1814,7 @@ func (rt *HandleT) generatorLoop(ctx context.Context) {
 	for {
 		select {
 		case <-ctx.Done():
+			rt.stopWorkers()
```

Comment:
I'm not sure about the way we're handling the start and stop here. The workers are initialized only once and that happens on `Setup()`, not on `Start()`. Then the `generatorLoop` (this method) is started by `Start()` and the `ctx` is canceled in `Shutdown()`. So correct me if I'm wrong but it looks like if we do:
1. setup
2. start
3. stop
4. start
5. stop // it breaks because we close a channel that was already closed?

So maybe along with this change it could make sense moving the `initWorkers` to `Start()`.

Reply:
Yeah this sounds correct actually, @lvrach Are we missing anything here?

Reply:
The way it currently works is a bit confusing, but if you look at https://github.com/rudderlabs/rudder-server/blob/5690a383abc4ce1fae93e8553d2b20819571369d/router/manager/manager.go#L67

our flow actually is:
1. setup
2. start
....
3. shutdown
4. setup
5. start
....

So after every shutdown we create a new router and call start, we don't reuse it.

Reply:
Resolving this for now, We can pick up the refactoring over a separate PR